src/components/flyout/deviceReconfigureFlyout.js

```diff
@@ -32,14 +42,46 @@ class DeviceReconfigureFlyout extends React.Component {
     this.commonConfigValueChanged = this.commonConfigValueChanged.bind(this);
     this.onChangeInput = this.onChangeInput.bind(this);
     this.applyDeviceConfigureJobsData =this.applyDeviceConfigureJobsData.bind(this);
+    this.checkJobStatus = this.checkJobStatus.bind(this);
   }
 
-  componentWillMount() {
-    this.calcCommonConfiguration(this.props.devices);
+  componentDidMount() {
+    const { devices, propertyUpdateJobs } = this.props;
+    this.checkJobStatus(devices, propertyUpdateJobs);
+    this.calcCommonConfiguration(devices);
   }
 
   componentWillReceiveProps(nextProps) {
-    this.calcCommonConfiguration(nextProps.devices);
+    const { devices, propertyUpdateJobs } = nextProps;
+    if (!_.isEqual(propertyUpdateJobs, this.props.propertyUpdateJobs)) {
+      this.checkJobStatus(devices, propertyUpdateJobs);
+    }
+    this.calcCommonConfiguration(devices);
+  }
+
+  checkJobStatus (devices, propertyUpdateJobs) {
+    const jobs = getRelatedJobs(devices, propertyUpdateJobs);
+    const jobStream = Rx.Observable.from(jobs);
+    jobStream
+      .flatMap(({ jobId, deviceIds }) =>
+        Rx.Observable
+          .fromPromise(ApiService.getJobStatus(jobId))
+          // Get completed jobs
+          .filter(({ status }) => status === 3)
+          .map(job => ({...job, deviceIds}))
```

Comment:
Doesn't look like you care about the job at this point so `.flat(_ => deviceIds)` should be enough. (if you flatMap an array it will have the same effect as doing a flatMap of Rx.Observable.from(array)). I would also break out of the first flatMap at this point.